### ISSUE TITLE
Static files not loaded properly when running behind nginx reverse proxy

### ISSUE THREAD

**Issue by WangCharlie** [aspnet/Home#2143]
i have created a web project with visual studio 2017 and deployed to centos 7 then host by nginx. 
now i use dotnet xxx.dll to start the app.
it wokrs on http://localhost:5000 port but not working on http://localhost.
here is the snapshot.
![image](https://user-images.githubusercontent.com/6106454/29443990-92b7e2ac-840e-11e7-94b1-2ebd3a0ad83c.png)

![image](https://user-images.githubusercontent.com/6106454/29444001-a1f78cb8-840e-11e7-8ac3-122863296528.png)

**Comment by natemcmaster** [comment]
Can you share the details of how the site is configured in nginx? I suspect it may be a misconfiguration of the reverse proxy settings.

**Comment by WangCharlie** [comment]
here is the config file of nginx
```
server
{
    listen 80;
    server_name 10.200.10.179;
    index index.php index.html index.htm default.php default.htm default.html;
    root /www/wwwroot/recognizer;
    
    location / {
        proxy_pass http://localhost:5000;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection keep-alive;
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }
    
    #SSL-START SSL相关配置，请勿删除或修改下一行带注释的404规则
    #error_page 404/404.html;
    #SSL-END
    
    #ERROR-PAGE-START  错误页配置，可以注释、删除或修改
    error_page 404 /404.html;
    error_page 502 /502.html;
    #ERROR-PAGE-END
    
    #PHP-INFO-START  PHP引用配置，可以注释或修改
    include enable-php-54.conf;
    #PHP-INFO-END
    
    #REWRITE-START URL重写规则引用,修改后将导致面板设置的伪静态规则失效
    include /www/server/panel/vhost/rewrite/10.200.10.179.conf;
    #REWRITE-END
    
    location ~ .*\.(gif|jpg|jpeg|png|bmp|swf)$
    {
        expires      30d;
        access_log off; 
    }
    
    location ~ .*\.(js|css)?$
    {
        expires      12h;
        access_log off; 
    }
    access_log  /www/wwwlogs/10.200.10.179.log;
}
```

**Comment by natemcmaster** [comment]
Try removing the "root" directive on line 6.

**Comment by WangCharlie** [comment]
remove the 'root'  and restart nginx .  it does still not work

![image](https://user-images.githubusercontent.com/6106454/29451534-9dcd105a-8435-11e7-830c-37ce26ae0731.png)

**Comment by natemcmaster** [comment]
Can you verify the reverse proxy is sending the request for js and css files through to the ASP.NET Core app?

**Comment by WangCharlie** [comment]
how to verify the reverse proxy is sending the request for js and css files? 
i can see imgs files is loaded properly but js and css files not

**Comment by WangCharlie** [comment]
it's very wired  that it works with js and css files reference out of site.
and imgs files are the same as js and css files was loaded from local.
here is the snapshot

![image](https://user-images.githubusercontent.com/6106454/29501100-7e10a694-8658-11e7-86a8-b338f55d4591.png)

**Comment by natemcmaster** [comment]
@WangCharlie enable logging from your ASP.NET Core app and check if the logs show requests for static files. If they do not show up, but other requests do, it probably means nginx is not reverse-proxying these requests into aspnet.

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging

**Comment by garthtee** [comment]
Hi, I'm having the same issue. How did you resolve this?

**Comment by mota57** [comment]
I have this problem as well. I Just forward the log as the following guide [microsoft asp.net core](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/linux-nginx?tabs=aspnetcore2x)

This is the how it looks my configuration. this the website... http://207.246.66.92/

server {
	listen 80;
	listen [::]:80;

	server_name 207.246.66.92;

	location / {
		try_files $uri $uri/ =404;
		  proxy_pass http://localhost:5000;
			proxy_http_version 1.1;
			proxy_set_header Upgrade $http_upgrade;
			proxy_set_header Connection keep-alive;
			proxy_set_header Host $http_host;
			proxy_cache_bypass $http_upgrade;
	}

}

**Comment by muneebzulfiqar** [comment]
Why was this issue closed? There is no resolution to this.

**Comment by Eilon** [comment]
**Hi, it looks like you are   posting on a closed issue/PR/commit!**

We're very likely to lose track of your bug/feedback/question unless you:

1. Open a new issue
2. Explain very clearly what you need help with
3. If you think you have found a bug, include detailed repro steps so that we can investigate the problem.

**Comment by AndriiZ** [comment]
Issue still not resolved (same issue on nginx 1.14.0 + asp.net core 2.1 on rpi)
